### ISSUE TITLE
Reorder versions

### ISSUE THREAD

**Issue by NatoBoram** [modrinth/knossos#307]
### Is your feature request related to a problem? Please describe.

When I discovered Modrinth, I published a single `.jar` of the latest version of my mod. Now that I want to use the platform a bit more, I published my older versions as well as the newest one. However, they are ordered by publishing date.

![Screen Shot 2021-09-25 at 23 11 39](https://user-images.githubusercontent.com/10495562/134792043-acab7287-69c2-462a-9d5f-082f776e3bcf.png)

### Describe the solution you'd like

It would be appropriate if mod authors could re-order the published versions as they see fit.

### Describe alternatives you've considered

Ideally, the default sort should be influenced by the semver version number rather than the publishing date.

**Comment by wafflecoffee** [comment]
> Ideally, the default sort should be influenced by the semver version number rather than the publishing date.

This idea is great in principle, but unfortunately, a lot of people... don't know what semver is. Even developers. :slightly_frowning_face: 

Also, some mods publish multiple different versions at different version numbers for different MC versions, which would lead to newer versions being published first no matter what. Take, for example, Mod Menu: it has a v1.16.x series for MC 1.16.5, a v2.0.x series for MC 1.17.1, and a v2.1.x series for MC 1.18 snapshots. It supports all three of these at the same time, but the 1.16.5 version would be buried underneath the multiple 1.17/1.18 versions if it were to be sorted by semver.

**Comment by NatoBoram** [comment]
Yeah, I had to unsubscribe from ModMenu's releases for that reason :D

I guess it's not so bad if their 1.16 versions are grouped together, it even makes more sense when I think about it. The new version still uses the old API, and if you're looking for the old API, you aren't really interested in releases for the new API.

Another problem is the suffix versions, like `1.0.2-dev` and `1.0.2-beta1`, `1.0.2-pre8`. There's not really any way to sort them out.

But anyway, the best way I think is just to let mod authors decide what they want.

**Comment by kencinder** [comment]
Ideally this should be up to the user to sort things how they want when looking for files, not the developer.

eg; You should be able to filter a projects versions page by minecraft version. I have no use seeing anything else when I'm looking for files that are for 1.18.1 for instance.

As it stands right now, there are zero filtering or sorting options on project pages, which really sucks.

**Comment by wafflecoffee** [comment]
@kencinder That's being tracked as #335 :)

**Comment by HyperSoop** [comment]
So since I was linked here, is any work happening regrading this
